x-pack/plugins/ingest_manager/server/services/epm/registry/index.ts

```diff
@@ -182,17 +182,20 @@ async function extract(
   onEntry: (entry: ArchiveEntry) => void
 ) {
   const archiveBuffer = await getOrFetchArchiveBuffer(pkgName, pkgVersion);
+  const archiveKey = getArchiveKey(pkgName, pkgVersion);
+  // shouldn't be possible since getOrFetchArchiveBuffer -> fetchArchiveBuffer sets the key
+  if (!archiveKey) throw new Error('no archive key');
+  const isZip = archiveKey.endsWith('.zip');
+  const libExtract = isZip ? unzipBuffer : untarBuffer;
 
-  return untarBuffer(archiveBuffer, filter, onEntry);
+  return libExtract(archiveBuffer, filter, onEntry);
 }
 
 async function getOrFetchArchiveBuffer(pkgName: string, pkgVersion: string): Promise<Buffer> {
-  // assume .tar.gz for now. add support for .zip if/when we need it
-  const key = getCacheKey(`${pkgName}-${pkgVersion}`);
-  let buffer = cacheGet(key);
+  const key = getArchiveKey(pkgName, pkgVersion);
+  let buffer = key && cacheGet(key);
   if (!buffer) {
     buffer = await fetchArchiveBuffer(pkgName, pkgVersion);
-    cacheSet(key, buffer);
```

Comment:
moved the caching from this `getOrFetch` function to inside the `fetchArchiveBuffer`